packages/cloudflare/src/api/do-sharded-tag-cache.ts

```diff
@@ -43,17 +82,56 @@ class ShardedD1TagCache implements NextModeTagCache {
     return durableObject.get(id);
   }
 
+  private generateRandomNumberBetween(min: number, max: number) {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+  }
+
+  /**
+   * This function generates an array for the double sharding
+   * @param tags The tags to generate shards for
+   * @param shardType Whether to generate shards for soft or hard tags
+   * @param generateAllShards Whether to generate all shards or only one
+   * @returns An array of shardId and tag
+   */
+  private generateShardArray(tags: string[], shardType: "soft" | "hard", generateAllShards = false) {
```

Comment:
nit: `{generateAllShards = false}` would be nicer than a bool (everywhere applicable)